Add   fonts,  rename logo Img, fixed bugs markUp

mike14112 opened the pull request:
Добавил  Fonts,   пофиксил  баги  в верстке.   переиминовал  название изображение   теперь на устройствах  выглядит  одинаково       как  в макете

mike14112 committed:
add   fonts,  rename logo Img, fixed bugs markUp

mike14112 committed:
fix:

fix markup

mike14112 committed:
feat: add wrapper to  tabs

mike14112 committed:
finished the layout, added a modal window, placing an order

karyna4k commented on index.html:
```diff
@@ -7,20 +7,61 @@
+          <form action="" class="form">
+        <h2>Доставка</h2>
+            <input type="text" placeholder="Ваше Имя"  class="input-name">
+            <input type="text" placeholder="Телефон" class="input-phone">
```
в поле телефон нет валидации, можно вписать что угодно (тип поля явно не текст)
https://prnt.sc/idNZrval9eq_

karyna4k commented on app.js:
```diff
@@ -165,50 +138,116 @@ const renderCart =  (products) => {
+  modal.classList.remove("hide");
+});
+
+btnClose.addEventListener("click", () => {
```
после закрытия модального окна не очищаются поля, а должны

karyna4k commented on app.js:
```diff
@@ -1,152 +1,131 @@
+});
 // кнопки корзины
 
+// кнопки добавление
```
добавления чего?

karyna4k commented on app.js:
```diff
@@ -1,152 +1,131 @@
+/// продукты
+const cards = document.querySelectorAll(".card");
+
+// onclick(btnTabs[i].style.backgroundColor = 'black')
```
не забывай удалять неиспользуемый код

karyna4k commented on app.js:
```diff
@@ -1,152 +1,131 @@
+  renderCart(products);
+};
 
 //  //показ-скрытие корзины   с помощью кнопки
```
зачем для этого два отдельных события, если ты используешь toggle?

karyna4k commented on app.js:
```diff
@@ -1,152 +1,131 @@
-  if(arr.length == 0){
-    mainCart.classList.add('hide')
-    cartHeader.textContent = ' В корзине пока пусто :( '
+//показ-скрытие корзины
```
подумай как это объединить с событиями выше, потому что слишком много слушателей создаешь

karyna4k commented on app.js:
```diff
@@ -159,50 +138,116 @@ const renderCart =  (products) => {
 // control Cart
-const  controlCart = ( arr, btnMinus, btnBtnPlus) => {
-
+const controlCart = (arr, btnMinus, btnBtnPlus) => {
```
плохая практика объединять внутри функции сразу несколько функций у которых разные цели

одна цель - одна функция

mike14112 replied:
Исправил

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+             <div class="delivery-section hide">
+              <input type="text" placeholder="Улица, дом, квартира" class="input-adress">
+               <div class="section-home-input">
+                <input type="text" placeholder="Этаж"  class="input-floor" >
```
почему input="text"?

этаж - это цифра

mike14112 replied:
Исправил

karyna4k commented on app.js:
```diff
@@ -159,50 +138,116 @@ const renderCart =  (products) => {
+
+  if (
+    !isDelivery &&
+    inputName.value.length >= 4 &&
```
для чего эта проверка?

mike14112 replied:
isDelivery  при выборе на радиокнопку меняется статус доставки самовывоз или доставка

karyna4k replied:
я про проверки длины

mike14112 replied:
Да кстати можно прям в инпуте   это  все зделать  что  я прям в js  поместил

mike14112 replied:
щас исправлю 
и продолжу исправлять

mike14112 replied:
Исправлено

karyna4k commented on app.js:
```diff
@@ -159,50 +138,116 @@ const renderCart =  (products) => {
+  if (
+    !isDelivery &&
+    inputName.value.length >= 4 &&
+    inputPhone.value.length >= 8
```
тот же вопрос что и выше

mike14112 replied:
Исправлено

karyna4k commented on app.js:
```diff
@@ -159,50 +138,116 @@ const renderCart =  (products) => {
 
+  if (
+    isDelivery &&
+    inputName.value.length >= 4 &&
```
с 231 по 235 строку что это???

mike14112 replied:
Проверка теперь  в инпуте

karyna4k reviewed (commented):
Add comments to the code.

Answer pls

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+    <header class="header">
+      <img src="img/logo.png" alt="" />
     </header>
     <div class="header-section">
```
неудачный нейминг, это же не хэдер

подумай как лучше назвать

karyna4k replied:
и нет контейнера внутри, на мобильной версии текст уезжает вправо

mike14112 replied:
Зделано как на Макете там текст не по центру   а чуток левее   
![2](https://github.com/mike14112/Meals/assets/89590104/d5be218a-4866-48a9-9196-b8f38b811693)
 
![3](https://github.com/mike14112/Meals/assets/89590104/fe527b6e-066b-462e-922c-2a4fcf941768)

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+                  <div class="cart-footer">
+                    <img src="img/icon/delivery.png" alt="" />
+                    <span>Бесплатная Доставка</span>
+                    <button class="btn-hide">Свернуть</button>
```
кнопка свернуть есть, а где кнопка развернуть, когда корзина свернута?

https://prnt.sc/5fFHi-s07QVX
https://prnt.sc/_qlnj-Jksih3

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+          <img src="img/close-modal.png" alt="" class="btn-close-modal">
+          <form action="" class="form">
+        <h2>Доставка</h2>
+            <input type="text" placeholder="Ваше Имя"  class="input-name">
```
а что за форма отправки без тега form?

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+        <div class="form-modal">
+          <img src="img/close-modal.png" alt="" class="btn-close-modal">
+          <form action="" class="form">
+        <h2>Доставка</h2>
```
заголовок можно сделать над формой, в форме он ни к чему

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
-       </article>
+             <button class="btn-confirn">Оформить</button>
+          </form>
+          <button class="btn-confirn-mobile">Оформить</button>
```
1. Зачем отдельная кнока confirm для мобильной версии?
2. Почему она за формой?

mike14112 replied:
ну если  смотреть  на  Мобильную  Верстку там кнопка ваабще вниз уежает  хотя очень плохо вышло подумаю надж Исправлением

mike14112 replied:
Иправлено

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
-            </div>
-       
-       </article>
+             <button class="btn-confirn">Оформить</button>
```
После клика по кнопке нужно скрывать модальное окно (общепринятая практика)

https://prnt.sc/CrZD8Y816_Sn

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
-           <button class="btn-add">Добавить</button>
+      <section class="navigation">
+        <div class="container">
+          <ul class="nav">
```
скролл выглядит не очень и к тому же мешает. есть специальные css свойства, которые его скрывают 

https://prnt.sc/XCR_exW9NP3p

mike14112 replied:
Спасибо большое   выполнено

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
-   </article>
-   
+      </section>
+      <section>
```
выглядит некрасиво, нужно исправить

https://prnt.sc/f_hV90Wu0D9r

и еще такое происходит

https://prnt.sc/8qsmKwcY7GeP

mike14112 replied:
Знаю  так задумано  Дизайнером

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+              <div class="tab-content" id="3">
+                <h2>Хот-доги</h2>
+                <div class="list-products">
+                  <li class="card">
```
таааак-с, с неймингом классов беда

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+                <div class="list-products">
+                  <li class="card">
+                    <img src="img/items/3-1.png" alt="" class="img" />
+                    <p class="price">290₽</p>
```
почему тег p, если это заголовок карточки?

советую использовать h3 или h4, исходя из иерархии

https://html5book.ru/razdely-dokumenta/#headings

mike14112 replied:
ЗДлано заменил все теги  на   h3-h5

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+                    <img src="img/items/3-1.png" alt="" class="img" />
+                    <p class="price">290₽</p>
+                    <span class="name">Домашний хот-дог</span>
+                    <div class="footer-card">
```
можно сделать без обертки, либо тогда оборачивать все содержимое кроме картинки

mike14112 replied:
хм не совсем понял  если честно....

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+        <div class="main-tablet-footer">
+          <div><img src="img/footer-logo.png" alt="" /></div>
+          <div>
+            <a href="tel:+79308333811">
```
не так как на макете https://prnt.sc/v6wb6TUAw3lt

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+        </div>
+        <div class="contact-section">
+          <h2>Номер для заказа</h2>
+          <div class="links-footer">
```
не так как на макете https://prnt.sc/v6wb6TUAw3lt

mike14112 replied:
В следущем Релизе   я поменял  верстку Footer      теперь эта одна вертска   и управляется Grid.  И поменял нейминг  автора Дизайна

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+            <a href="https://vk.com/"><img src="img/icon/vk.png" alt="" /></a>
+          </div>
+        </div>
+        <span class="autor-m"
```
что за странное название класса "autor-m"?

mike14112 replied:
Автор  Дизайнера подумаю над другим неймингом    -m  обозначает  для телефонов этот  тэг

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+        >
+      </div>
 
+      <div class="tablet-footer">
```
для планшетов не нужно делать отлельную верстку, можно сделать без этого. нужно переделать

mike14112 replied:
решено

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
-  <div class="main-footer">
-    <img src="img/footer-logo.png" alt="">
-    <span class="autor">© YouMeal, 2022 <br>
+      <div class="mobile-footer">
```
для мобильных не нужно делать отлельную верстку, можно сделать без этого. нужно переделать

mike14112 replied:
решено

karyna4k commented on index.html:
```diff
@@ -1,351 +1,425 @@
+         
+        </div>
+       
+        <div class="group-links-tablet">
```
для планшетов не нужно делать отлельную верстку, можно сделать без этого. нужно переделать

mike14112 replied:
решено

mike14112 commented on app.js:
```diff
@@ -1,5 +1,5 @@
 //// кнопки  tab
```
хорошо

karyna4k replied:
и не убрал :)

mike14112 commented on main.css:
```diff
@@ -50,15 +56,15 @@ header{
  background-color: #FFAB08;
-border-radius: 0 0  50rem 50rem;
-
+border-radius: 0 0  20rem 15rem;
```
Хорошо Исправлю

mike14112 replied:
Исправлено

mike14112 committed:
Refactor:  changed the layout of the footer, rename logo

Refactor:

 removed z-index and
 overflow - auto

replaced rem with px in border-radius

Replaced with English comment in js

mike14112 committed:
Refactor:  change type number  &  added conditions to the input

Refactor :

replaced the input type where the Number type was used.
Condition added Directly to Input

removed all unnecessary conditions in js

mike14112 committed:
Fix:  Products do not disappear when you click on the checkout button

mike14112 committed:
Fixed: Fixed now the modal window does not close when the fields are empty

Fixed:  added conditions to js if the fields are empty the conditions are not met

mike14112 committed:
Fixed: fixed a bug in css styles are not applied

mike14112 committed:
Refactor: modal  refactor

Refactor :  delete confirn-btn  with mobile
add div  group-modal
   deleted test eventlistner

mike14112 committed:
Refactoring: change className by tage

Refactoring :
split the control cart function,
changed the names of classes for tags,
in the card I changed the tags instead of (p, span) now the hierarchy is h3-h5

mike14112 commented on index.html:
```diff
@@ -9,19 +9,23 @@
 
     <header>
-        <img src="img/HeaderLogo.png" alt="">
+        <img src="img/logo.png" alt="">
```
Исправлено

karyna4k replied:
я говорила не переименование картинки, а про структуру самого лого

хэдер -> внутри ссылка на главную -> внутри картинка

просто картинкой не верстается, вот пример
```
<div class="logo">
  <a href="ссылка_на_главную">
   <img src="img/logo.png" />
  </a>
</div>
```

mike14112 commented on main.css:
```diff
@@ -11,6 +14,9 @@ body{
     background-color:#F2F2F3;
     overflow-x: hidden;
     z-index: 0;
```
Спасибо. Исправил

mike14112 commented on index.html:
```diff
@@ -1,351 +1,425 @@
+                </div>
+              </div>
+              </div>
+  
```
Исправлено

mike14112 committed:
Refactoring : placed an h2 tag behind the form

karyna4k commented on app.js:
```diff
@@ -21,22 +21,28 @@ const  cartHeader = document.querySelector('.cart-header-empty'),
 /// продукты 
```
и не удалил :)

karyna4k commented on app.js:
```diff
@@ -1,152 +1,125 @@
+  btnHIde = document.querySelector(".btn-hide");
+const btnCheckout = document.querySelector(".btn-checkout");
+
+/// Items
+const cards = document.querySelectorAll(".card");
```
достаточно двух наклонных черточек, больше не нужно

karyna4k commented on main.css:
```diff
@@ -1,589 +1,606 @@
+  }
+
+  .promo-group {
+    border-radius: 0 0 10rem 10rem;
```
снова ремы, исправь плз
